### DISCUSSION TITLE
does ventoy works with UEFI by default or should it be set manually ?

### DISCUSSION THREAD

**Discussion by a-ovi** [ventoy/Ventoy#2951] [Q&A] [answered]
first of all, I'm not that much familiar with BIOS/Legacy/UEFI, so pardon me when I misreference these terms.

**what I want to achieve:** currently my BIOS mode is "Legacy", but I want to make sure that, when I freshly install Manjaro (with ventoy of course), it should be in "UEFI" mode.

**the problem I'm facing:** If I change BIOS mode from "Legacy" to "UEFI", and then try to boot from the ventoy, then "verification failed: (0x1A) security violation" shows,
![1](https://github.com/user-attachments/assets/0be5198c-90cc-4054-9bf6-1a1502d4f281)
and clicking on "ok" shows "Shim UEFI key management"
![2](https://github.com/user-attachments/assets/3a4e97ad-f65a-4897-987b-ce7d3feba225)
and then pressing any keys leads to "Perform MOK management" .
![3](https://github.com/user-attachments/assets/38ce859c-e91a-4cac-a7f7-5f5327382942)
If I press "continue boot" , the first image appears and creates a loop. and i don't know what to do with other two options.

**But if I disable secure boot** from BIOS, then everything works fine and I can boot the live images.

**My question:** is there any way to do the "UEFI" thing with ventoy, while secure boot is enabled? or these two options (UEFI+secure boot AND ventoy) doesn't blend together?  
what options should I check/uncheck while installing ventoy on a USB so that it supports UEFI BIOS mode? is Selecting "GPT" enough for UEFI?

**Comment by steve6375** [comment, marked as answer]
Yes, you use mok manager to enroll a key.
Read the document pages on the website 
The key file is in the root of the 2nd efi partition.

**Reply by a-ovi** [reply]
fixed the problem from the [official docs](https://www.ventoy.net/en/doc_secure.html) , thank you very much.

just one thing to be clear about: So now If I switch to `UEFI` in `BIOS mode` and then try installing OS (e.g. manjaro) freshly from ventoy usb, will it be on `UEFI`? or I have to do some manual works?

**Reply by steve6375** [reply]
If you want to boot the target system from it's internal hard drive in UEFI mode, then you should boot to Ventoy and install the OS in UEFI mode. Secure Boot can be either enabled or disabled when you boot to Ventoy. If it was disabled, you can still secure boot to the installed hard disk just fine (as long as that OS supports secure booting).
If you install from Ventoy in legacy\BIOS mode, you can boot the final OS on the hard disk in legacy/BIOS mode.
Some installation processes may allow you to boot the final OS in both legacy and UEFI modes, but this is not very common. UEFI requires one of the partitions to be FAT-based, so if a legacy installation is performed and no FAT partition is present, UEFI booting will not be possible (although some UEFI firmware can also UEFI-boot from an NTFS partition as well as a FAT-based partition).

**Reply by a-ovi** [reply]
thank you
